composer.json

```diff
@@ -20,6 +20,7 @@
         "php": "~8.2.0 || ~8.3.0 || ~8.4.0",
         "ext-openssl": "*",
         "ext-sodium": "*",
+        "phpseclib/phpseclib": "^3.0.36",
```

Comment:
Here I disagree with @lcobucci https://github.com/lcobucci/jwt/issues/1074#issuecomment-2486585502
While `phpseclib/phpseclib` is indeed good, I don't like adding that entire dependency for such and uncommon and superseded signature algorithm.

Reply:
We can disagree, and I totally see your point (specially for this use case). 

I don't like adding dependencies but, sometimes, they can help us going further with reduced effort.

Can we quantify things to have a more complete analysis? What are the pros/cons? How much is the lib footprint growing with each approach? How is our maintenability affected?

Reply:
My concern is that as a user I have no pratical way to tell if a flaw or security issue on the dependency directly affects (or not) the part of the main library I use.

(Native extensions have been less affected by this problem so far.)

Forcing every user to install such a huge dependency for something they will likely not use is annoying:

```json
    "replace": {
        "symfony/polyfill-ctype": "*",
        "symfony/polyfill-iconv": "*",
        "symfony/polyfill-intl-grapheme": "*",
        "symfony/polyfill-intl-idn": "*",
        "symfony/polyfill-intl-normalizer": "*",
        "symfony/polyfill-mbstring": "*",
        "symfony/polyfill-php70": "*",
        "symfony/polyfill-php72": "*",
        "symfony/polyfill-php73": "*",
        "symfony/polyfill-php80": "*",
        "symfony/polyfill-php81": "*",
        "symfony/polyfill-php82": "*",
        "symfony/polyfill-php83": "*"
    },
```

I have no scientific threshold of (dep weight) to (benefit gained) ratio to respect.

But I can tell you this is the exact case I'd publish a separate `jwt-rssassa-pss` package for.

Reply:
@Slamdunk I wonder what that list of replacements is supposed to say. It does not seem to be anything that is added by PhpSecLib.

Anyways, so far I had some fun doing hands-on work, and I admit I'm not the person promoting this schema, and I'm not pushing it getting merged - so whatever decision is made is fine by me.

Keep in mind though, that some keywords for longstanding feature requests do appear in PhpSecLib, like JWK. And in fact the documentation explicitly uses JWT as an example for signing the tokens, although in quite a low-level approach. It could power all schemas that are to offer here, maybe without Blake2B.

If you decide to move on, I'll start fighting the mutants next week (which didn't appear locally - I was running `make`, no clue what happens behind the scene for now, but will look into it).